server/clip_server/executors/clip_torch_dataloader.py

```diff
@@ -0,0 +1,92 @@
+import io
+import os
+from time import perf_counter
+from typing import TYPE_CHECKING, Optional, List
+
+import torch
+from PIL import Image
+from jina import Executor, requests
+from torch.utils.data import Dataset, DataLoader
+
+from clip_server.model import clip
+
+if TYPE_CHECKING:
+    from docarray import DocumentArray, Document
+
+
+class DocumentDataset(Dataset):
+    def __init__(self, device):
+        self.device = device
+        self.da = None
+
+    def set_da(self, da: 'DocumentArray'):
+        self.da = da
+
+    def __len__(self):
+        return len(self.da)
+
+    def __getitem__(self, idx):
+        text = self.da[idx].text
+        tensor = clip.tokenize(text)
+        return tensor[0]
+
+
+class CLIPEncoder(Executor):
+    def __init__(
+            self,
+            name: str = 'ViT-B/32',
+            device: Optional[str] = None,
+            jit: bool = False,
+            batch_size: int = 1000,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        if not device:
+            self._device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        else:
+            self._device = device
+        self._model, self._preprocess = clip.load(name, device=self._device, jit=jit)
+        self.batch_size = batch_size
+
+        self.dataset = DocumentDataset(self._device)
+        self.dataloader = DataLoader(self.dataset, batch_size=self.batch_size, num_workers=48, persistent_workers=True)
```

Comment:
Can we confirm the performance diff doesn't come from `num_workers=48`?

Reply:
it does not mean more workers will definitely lead to faster processing speed. Generally, it should not exceed the number of CPU cores we have.

Reply:
we have 64 cpu core in the berlin machine. In our case we have to prepare 10 batch so we could just stick with 10 workers

Reply:
If the improvement mainly comes from `num_workers=48`, then we can utilize `da.apply(..., num_workers=48)` implemented in `docarray` instead.